### PR TITLE
ViewPagerAndroid: FIX folly::toJson: JSON object value was a NaN or INF

### LINKED ISSUE #9750: ViewPagerAndorid RCTEventEmitter.receiveEvent Error bug
### Issue Description

Error calling function:ViewPagerAndorid RCTEventEmitter.receiveEvent

### Steps to Reproduce / Code Snippets
```

<View>
<ViewPagerAndorid />
</View>
```

# 

### Additional Information

* React Native version: [0.32.0]
* Platform(s) : Android
* Operating System (macOS, Linux, or Windows?):macOS

### PR DESCRIPTION
**Pull request by leeight**
Explain the **motivation** for making this change. What existing problem does the pull request solve?

Under certain scenario, `PageScrollEvent.offset` was initialized to `NaN`, which cause `folly::toJson` failed, and FIX #9750 

<https://github.com/android/platform_frameworks_base/blob/e71ecb2c4df15f727f51a0e1b65459f071853e35/core/java/com/android/internal/widget/ViewPager.java#L1689>

image

**Test plan (required)**

<http://stackoverflow.com/questions/39327429/reactnative-viewpagerandroid-rcteventemitter>

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
By analyzing the blame information on this pull request, we identified @spicyj and @davidaurelio to be potential reviewers.

File: Libraries/Components/ViewPager/ViewPagerAndroid.android.js [hunk 1]
```diff
@@ -141,7 +141,7 @@ class ViewPagerAndroid extends React.Component {
   };
 
   componentDidMount() {
-    if (this.props.initialPage) {
+    if (this.props.initialPage != null) {
```
Comment (andreicoman11) [inline Libraries/Components/ViewPager/ViewPagerAndroid.android.js]: Why is this necessary?

**Comment by andreicoman11** [comment]
Lgtm, just an inline

File: Libraries/Components/ViewPager/ViewPagerAndroid.android.js (hunk 1 above)
Comment (leeight) [inline Libraries/Components/ViewPager/ViewPagerAndroid.android.js]: `<ViewPagerAndroid initialPage={0} />`

**Comment by facebook-github-bot-3** [comment]
@leeight updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9755/files/9c376f082d4fea88dc598e8a905afa4931b8a371..0eb1acdf471abb1cd4231005c678b148459a08f2)

**Comment by facebook-github-bot-3** [comment]
@leeight updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9755/files/0eb1acdf471abb1cd4231005c678b148459a08f2..9d014f425c2347e1041c9455c8213d907bfe4c80)

**Comment by leeight** [comment]
Ping @andreicoman11 @davidaurelio

File: Libraries/Components/ViewPager/ViewPagerAndroid.android.js (hunk 1 above)
Comment (andreicoman11) [inline Libraries/Components/ViewPager/ViewPagerAndroid.android.js]: I don't think this is needed here.

File: ReactAndroid/src/main/java/com/facebook/react/views/viewpager/PageScrollEvent.java [hunk 2]
```diff
@@ -34,7 +34,10 @@
   protected PageScrollEvent(int viewTag, int position, float offset) {
     super(viewTag);
     mPosition = position;
-    mOffset = offset;
+
+    // folly::toJson default options don't support serialize NaN or Infinite value
+    mOffset = (Float.isInfinite(offset) || Float.isNaN(offset))
```
Comment (andreicoman11) [inline ReactAndroid/src/main/java/com/facebook/react/views/viewpager/PageScrollEvent.java]: Doesn't this fit in one line?

**Comment by mkonicek** [comment]
This looks good, thanks for the fix!

I agree with @andreicoman11 we should keep `if (this.props.initialPage)` - it's more defensive programming as it also catches `undefined` besides `null`.

The Java code looks good. If you address the minor thing with keeping the original check in JS I think we should merge this.

**Comment by facebook-github-bot-1** [comment]
@leeight updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9755/files/9d014f425c2347e1041c9455c8213d907bfe4c80..2b7482bda53dda9089b9de045ea215958d916dac)

**Comment by leeight** [comment]
@mkonicek @andreicoman11 

But if initialPage was initialized by `0`, `if (this.props.initialPage)` was not our expected.

**Comment by andreicoman11** [comment]
Is there anything wrong with having `initialPage` to be 0?

**Comment by leeight** [comment]
I mean `this.setPageWithoutAnimation` will not be called if `initialPage` was set to 0.

**Comment by andreicoman11** [comment]
Ah, ok, that makes sense now, didn't understand that from your previous explanation. Feel free to add that back, then we can merge this.

**Comment by facebook-github-bot-7** [comment]
@leeight updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9755/files/2b7482bda53dda9089b9de045ea215958d916dac..f6a2268f4833d807793ffd83226f8112190f4b45)

**Comment by leeight** [comment]
Done

**Comment by andreicoman11** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-9** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3841674) to review internal test results.
